src/normalize.js

```diff
@@ -92,21 +94,65 @@ function extractAllLinks(json, uri, opts) {
   return ret;
 }
 
+function extractToVirtualKey(uri, rel, content, contentKey) {
+  const ret = {};
+  const virtualKey = `${uri}#${rel}`;
+  ret[virtualKey] = {
+    [contentKey]: content,
+  };
+
+  ret[uri] = {};
+  ret[uri][rel] = {
+    href: virtualKey,
+    virtual: true,
+  };
+  return ret;
+}
+
 function mergeEmbeddedStandaloneCollections(embedded, links, opts) {
   const ret = {};
   merge(ret, links);
   merge(ret, embedded);
 
   keys(embedded).forEach((uri) => {
+    // check all embedded properties for embedded collections
     keys(embedded[uri]).forEach((rel) => {
-      if (Array.isArray(embedded[uri][rel]) && uri in links && rel in links[uri]) {
-        ret[uri][rel] = links[uri][rel];
-        ret[links[uri][rel].href] = {
-          [opts.embeddedStandaloneListKey]: embedded[uri][rel],
-          [opts.metaKey]: { self: links[uri][rel].href },
-        };
+      if (Array.isArray(embedded[uri][rel])) {
+        // standalone link provided (store embedded list as standalone link)
+        if (uri in links && rel in links[uri] && isSingleLink(links[uri][rel])) {
+          ret[uri][rel] = links[uri][rel];
+          ret[links[uri][rel].href] = {
+            [opts.embeddedStandaloneListKey]: embedded[uri][rel],
+            [opts.metaKey]: { self: links[uri][rel].href },
+          };
+        } else if (opts.embeddedStandaloneListVirtualKeys
+          && rel !== opts.embeddedStandaloneListKey) {
+          // no standalone link provided --> generate virtual key
+          delete ret[uri][rel];
```

Comment:
I don't fully understand why this `delete` is necessary. Can you explain? How can there be something stored there already? Is this just to mirror the short duplicated code segment on line 148?

Reply:
Example from the test 'embedded list with link array generates virtual key':

When reaching this line, we have the following value for `ret` (populated in line 97/98):
```javascript
ret = {
  "http://example.com/posts/2620": {
    questions: [
      {
        href: "http://example.com/questions/295",
      },
    ],
  },
  "http://example.com/questions/295": {
    id: 295,
    text: "Why?",
    _meta: {
      expiresAt: 1513868982,
      self: "http://example.com/questions/295",
    },
  },
}
```

We know want to remove `ret['http://example.com/posts/2620']['questions']`, because we want to replace this array with our virtual link object. If we don't delete it here, we end up with this after merging the next line:
```
ret['http://example.com/posts/2620']['questions'] = 
{
  0: {
    href: 'http://example.com/questions/295',
  },
  href: 'http://example.com/posts/2620#questions',
  virtual: true,
}
```